protocol/binary_test.go

```diff
@@ -132,6 +133,30 @@ func TestBoolEOFFailure(t *testing.T) {
 	checkEOFError(t, wire.TBool, tests)
 }
 
+func TestStreamBool(t *testing.T) {
+	tests := []struct {
+		value bool
+	}{
+		{value: false},
+		{value: true},
+	}
+
+	for _, test := range tests {
+		var buff bytes.Buffer
+
+		// Encode with Streaming protocol
+		w := binary.BorrowWriter(&buff)
+		err := w.WriteBool(test.value)
+		assert.NoError(t, err)
+		binary.ReturnWriter(w)
+
+		// Decode with wire representation and assert that the value is the same
+		val, err := Binary.Decode(bytes.NewReader(buff.Bytes()), wire.TBool)
+		assert.NoError(t, err)
```

Comment:
`assert.NoError`'s like these should probably be from `require`.  The thought here is that `val` must be set (it could be `nil`) before continuing since any usage following this could get a "nil pointer exception".